MeshKernelNET/test/MeshKernelNETCoreTest/Api/MeshKernelCurvilinearTest.cs

```diff
@@ -0,0 +1,884 @@
+﻿using MeshKernelNETCore.Api;
+using NUnit.Framework;
+
+namespace MeshKernelNETCoreTest.Api
+{
+    [TestFixture]
+    [Category("MeshKernelNETCurvilinearTests")]
+    public class MeshKernelCurvilinearTest
+    {
+        public static DisposableCurvilinearGrid CreateCurvilinearGrid(
+            int numbOfRows = 3,
+            int numbOfColumns = 3,
+            double cellWidth = 1.0,
+            double cellHeight = 1.0)
+        {
+            var result = new DisposableCurvilinearGrid(numbOfRows, numbOfColumns);
+
+            int nodeIndex = 0;
+            for (int i = 0; i < result.NumM; ++i)
+            {
+                for (int j = 0; j < result.NumN; ++j)
+                {
+                    result.NodeX[nodeIndex] =  i * cellWidth;
+                    result.NodeY[nodeIndex] =  j * cellHeight;
+                    nodeIndex++;
+                }
+            }
+
+            return result;
+        }
+
+        [Test]
+        public void CurvilinearMakeUniformThroughAPI()
+        {
+            // Setup
+            using (var api = new MeshKernelApi())
+            using (var disposableGeometryList = new DisposableGeometryList())
+            {
+                var id = 0;
+                try
+                {
+                    id = api.AllocateState(0);
+
+                    var makeGridParameters = MakeGridParameters.CreateDefault();
+
+
+                    makeGridParameters.GridType = 0;
+                    makeGridParameters.NumberOfColumns = 3;
+                    makeGridParameters.NumberOfRows = 3;
+                    makeGridParameters.GridAngle = 0.0;
+                    makeGridParameters.OriginXCoordinate = 0.0;
+                    makeGridParameters.OriginYCoordinate = 0.0;
+                    makeGridParameters.XGridBlockSize = 10.0;
+                    makeGridParameters.YGridBlockSize = 10.0;
+                    makeGridParameters.UpperRightCornerXCoordinate = 0.0;
+                    makeGridParameters.UpperRightCornerYCoordinate = 0.0;
+
+                    Assert.IsTrue(api.CurvilinearMakeUniform(id, makeGridParameters, disposableGeometryList));
+
+                    var curvilinearGrid = new DisposableCurvilinearGrid();
+                    var success = api.CurvilinearGridGetData(id, out curvilinearGrid);
+                    Assert.IsTrue(success);
+
+                    Assert.AreEqual(4, curvilinearGrid.NumM);
+
+                    curvilinearGrid.Dispose();
```

Comment:
Move the Dispose calls to the finally clause. If one of the Assertions fail this will not be called.